test/NuGet.Core.Tests/NuGet.Protocol.Tests/LocalResourceTests/LocalPackageSearchResourceTests.cs

```diff
@@ -586,18 +587,53 @@ public async Task LocalPackageSearchResource_FileSourceAsync()
         }
 
         [Fact]
-        public async Task LocalPackageSearch_SearchAsync_WithCancellationToken_ThrowsAsync()
+        public async Task LocalPackageSearch_SearchAsync_WithCancellationToken_ImmediatelyThrowsAsync()
         {
             using (var root = TestDirectory.Create())
             {
+                // Arrange
                 var localResource = new FindLocalPackagesResourceV2(root);
                 LocalPackageSearchResource resource = new LocalPackageSearchResource(localResource);
 
-                await Assert.ThrowsAsync<OperationCanceledException>(
+                // Act & Assert
+                await Assert.ThrowsAsync<TaskCanceledException>(
                     async () => await resource.SearchAsync("", null, 0, 1, NullLogger.Instance, new CancellationToken(canceled: true)));
             }
         }
 
+        [Fact]
+        public async Task LocalPackageSearch_SearchAsync_SlowLocalRepository_WithCancellationToken_ThrowsAsync()
+        {
+            using (var pathContext = new SimpleTestPathContext())
```

Comment:
Can you use the inline `using` here as well?

Reply:
Sure.